src/TriggerApplication.cpp

```diff
@@ -0,0 +1,509 @@
+    //queueCstTeeToBufferObj.set_by_val<uint32_t>("capacity", tcQueueDesc->get_capacity());
+
+    //// Queue to the MLT
+    //// TODO: This is going to be wrong for now!
+    //// TODO: I think there can only be ONE TC queue to MLT, and that queue will have multiple inputs.
+    //// TODO: Should define that queue right at the top with the MLT.
+    //std::string cstTeeToMLTQueueUid("cst-tctee-to-mlt-");
+    //oksdbinterfaces::ConfigObject queueCstTeeToMLTObj;
+    //confdb->create(dbfile, "Queue", cstTeeToMLTQueueUid + UID(), queueCstTeeToMLTObj);
+    //queueCstTeeToMLTObj.set_by_val<std::string>("data_type", tcQueueDesc->get_data_type());
+    //queueCstTeeToMLTObj.set_by_val<std::string>("queue_type", tcQueueDesc->get_queue_type());
+    //queueCstTeeToMLTObj.set_by_val<uint32_t>("capacity", tcQueueDesc->get_capacity());
+
+    //// The TCBuffer
+    //std::string cstBufferUid("cst-tcbuffer");
+    //oksdbinterfaces::ConfigObject tcBufferCstObj;
+    //confdb->create(dbfile, "TCBuffer", cstBufferUid + UID(), tcBufferCstObj);
+
+    //// TODO: This is very bad, TimingTimingCandidate should either have it's own buffer schema relation, or the conf should be taken from the application-buffer relationship (if any?)
+    //auto tcBufferConf       = rndTCMakerConf->get_tcbuffer_conf();
+    //auto tc_buffer_conf_obj = tcBufferConf->config_object();
+    ////auto tcBufferLatency    = tcBufferConf->get_latencybuffer();
+    ////auto tcBufferReqHandler = tcBufferConf->get_requesthandler();
+
+    //tcBufferCstObj.set_obj("configuration", &tc_buffer_conf_obj);
+
+    //tcBufferCstObj.set_objs("inputs", {&queueCstTeeToBufferObj});
+
+    //// Push everything to modules & MLT queues
+    //modules.push_back(confdb->get<TCBuffer>(cstBufferUid + UID()));
+
+    //tcTeeCstObj.set_objs("outputs", {&queueCstTeeToBufferObj, &queueCstTeeToMLTObj});
+    //mlt_inputs.push_back(std::move(&queueCstTeeToMLTObj));
+
+    //// Push all the modules related with the RTCM
+    modules.push_back(confdb->get<CustomTriggerCandidateMaker>(cstTCMakerUid));
+    //modules.push_back(confdb->get<TCSetTee>(cstTCTeeUid + UID()));
+  }
+
+  // TODO: Warning! Not finished yet, remove and move down.
+  mltObj.set_objs("inputs", mlt_inputs);
+  mltObj.set_objs("outputs", mlt_outputs);
+
+  modules.push_back(confdb->get<ModuleLevelTrigger>(mltUid));
+
+  /**************************************************************
+   * Get the trigger graph
+   **************************************************************/
+  // Don't bother building the entire trigger graph if we have no tpstreams in.
+  if(!nReadoutLinks)
+    return modules;
+
+  // Get the triggering algos handler
+  auto trigger_algs = get_trigger_algs();
+  // Don't build the rest if we don't have triggering algs
+  if(trigger_algs.size() == 0)
+    return modules;
+
+  // Make the TPChannelFilters
+  auto tpChannelFilterConf = get_tpchannelfilter_conf();
```

Comment:
Not sure what you're doing with the TPChannelFilterConf, I suppose this is still a work in progress

Reply:
Yes, tpChannelFilter is very much work in progress, and will be probably completely changed in the future